### PR TITLE
Constant in select clause solved.

### PR DESCRIPTION
**Pull request by rolex4301**

### CODE REVIEW

File: mysite/unmasque/src/core/groupby_clause.py [hunk 1]
```diff
@@ -66,6 +67,15 @@ def doExtractJob(self, query):
                 self.see_d_min()
                 new_result = self.app.doJob(query)
 
+                #checking for 1 as constant in select clause.
+                for i in range(len(self.projected_attribs)):
+                    if self.projected_attribs[i] == '' :
+                        for j in range(1, len(new_result)):
+                            if new_result[j][i] != '1':
+                                check_array [i] = 2
```
Comment (ahanapradhan) [inline mysite/unmasque/src/core/groupby_clause.py:75]: I suppose these 1, 2 values are representative of something, some meaning. Create corresponding variable names as class constants and use them. E.g. CONSTANT_PROJECTION = 1. Then use CONSTANT_PROJECTION wherever 1 is used. This is similar to the use of "enum".

Comment (rolex4301) [reply mysite/unmasque/src/core/groupby_clause.py:75]: Ok ma'am will do it.

File: mysite/unmasque/src/core/projection.py [hunk 2]
```diff
@@ -107,10 +111,27 @@ def find_projection_dependencies(self, query, s_values):
...
+                    # No dependency and value is one so in groupby file will differentiate between count or 1.
+                    projected_attrib.append('')
 
         return projected_attrib, projection_names, projection_dep, True
 
+    def get_projection_column_type(self, query, projection_names):
+        mod_query = query.replace(";",'')
+        datatype_names = []
+        for i in projection_names:
+            typecheck_query = str('SELECT pg_typeof(' + i + ') AS data_type FROM (' + mod_query + ') AS subquery limit 1;')
```
Comment (ahanapradhan) [inline mysite/unmasque/src/core/projection.py:127]: later move it to postgres queries class. and create abstract method for it in common queries. These classes should not have anything specific to db engine.

**Review by ahanapradhan — APPROVED** [review]
as of now OK to merge. Can take the suggested changes later.

**Pull request merged**